app/models/user.rb

```diff
@@ -604,6 +632,24 @@ def saw_notification_id(notification_id)
     end
   end
 
+  def bump_last_seen_reviewable!
+    query = Reviewable.unseen_list_for(self, preload: false)
+
+    if last_seen_reviewable_id
+      query = query.where("id > ?", last_seen_reviewable_id)
+    end
+    max_reviewable_id = query.maximum(:id)
+
+    if max_reviewable_id
+      update!(last_seen_reviewable_id: max_reviewable_id)
+      MessageBus.publish(
+        "/reviewable_counts",
+        { unseen_reviewable_count: self.unseen_reviewable_count },
+        user_ids: [self.id]
+      )
```

Comment:
O I think it'll also be useful to extract this into a method so that you don't have to duplicate the MessageBus channel in multiple places. Although you might need to have some sort of conditional for the payload to be published.

Reply:
done!